### DISCUSSION TITLE
Can I listen & update model object

### DISCUSSION THREAD

**Discussion by hsul4n** [rrousselGit/riverpod#1904] [Q&A] [answered]
I'am wondering, Can I listen for property changes in a model class for example:

```dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

class Model {
  String foo = 'foo';
}

final fooProvider = StateProvider.autoDispose((ref) => Model());

class ExamplePage extends ConsumerStatefulWidget {
  const ExamplePage({super.key});

  @override
  ConsumerState<ExamplePage> createState() => ExamplePageState();
}

class ExamplePageState extends ConsumerState<ExamplePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Text(
        ref.watch(fooProvider).foo,
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          ref.read(fooProvider.notifier).update((state) => state..foo = 'bar');
        },
      ),
    );
  }
}
```

Changing whole object work fine, but changing object in model doesn't listen to changes.

**Comment by rrousselGit** [comment, marked as answer]
`update` expects the state is expected to be immutable.  
If you want to modify one property,  you should your object with only that property changed.

Although to begin with, as the docs of `StateProvider` mentions, for complex objects you probably should use `StateProvider` and instead use `StateNotifierProvider`.

**Reply by hsul4n** [reply]
@rrousselGit `StateNotifierProvider` will be the solution, but is there any chance for implementing solution so `ref.watch(fooProvider.notifier).foo` can listen for property changes.

**Reply by rrousselGit** [reply]
You're not supposed to do that. So no.
